### PR TITLE
feat: ZAP executor

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 

https://github.com/kubeshop/testkube/issues/1157

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: contrib/executor/zap/pkg/runner/runner.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	for key, value := range execution.Envs {
+		os.Setenv(key, value)
+	}
+
+	// when using file based ZAP parameters it expects a /zap/wrk directory
+	// we simply symlink the directory
+	os.Symlink(workingDir, filepath.Join(r.ZapHome, "wrk"))
+
+	output.PrintLogf("%s Running ZAP test", ui.IconMicroscope)
+	logs, err := executor.Run(r.ZapHome, scriptName, envManager, args...)
```
Comment (vsukhin) [inline contrib/executor/zap/pkg/runner/runner.go:117]: ideally, we need to prepare a command call from command and arguments defined in executor spec. you can check other executors. Sometimes, this is just direct parameters, in many cases, it's a template param, that we replace by actual value in executor

File: contrib/executor/zap/pkg/runner/runner.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+	output.PrintLogf("%s Preparing variables", ui.IconWorld)
+	envManager := env.NewManagerWithVars(execution.Variables)
+	envManager.GetReferenceVars(envManager.Variables)
+	// simply set the ENVs to use during execution
+	for _, env := range execution.Variables {
+		os.Setenv(env.Name, env.Value)
+	}
+
+	// convert executor env variables to runner env variables
```
Comment (vsukhin) [inline contrib/executor/zap/pkg/runner/runner.go:107]: techically, we already do this in api

File: contrib/executor/zap/pkg/runner/runner.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	output.PrintLogf("%s Preparing reports folder", ui.IconFile)
+	reportFile := filepath.Join(reportFolder, fmt.Sprintf("%s-report.html", execution.TestName))
+	scriptName := zapScript(scanType)
+	args := zapArgs(scanType, options, reportFile)
+
+	output.PrintLogf("%s Preparing variables", ui.IconWorld)
+	envManager := env.NewManagerWithVars(execution.Variables)
+	envManager.GetReferenceVars(envManager.Variables)
+	// simply set the ENVs to use during execution
+	for _, env := range execution.Variables {
```
Comment (vsukhin) [inline contrib/executor/zap/pkg/runner/runner.go:103]: we did this in api

File: contrib/executor/zap/pkg/runner/runner.go (hunk 1 above)
Comment (vsukhin) [reply contrib/executor/zap/pkg/runner/runner.go:117]: Need to print command and args

**Review by vsukhin — APPROVED** [review]
good job, a few notes

**Review by vsukhin — APPROVED** [review]
good job, a few notes

File: contrib/executor/zap/pkg/runner/runner.go (hunk 1 above)
Comment (vLia) [reply contrib/executor/zap/pkg/runner/runner.go]: added

File: contrib/executor/zap/pkg/runner/runner.go (hunk 3 above)
Comment (vLia) [reply contrib/executor/zap/pkg/runner/runner.go]: removed

File: contrib/executor/zap/pkg/runner/runner.go (hunk 2 above)
Comment (vLia) [reply contrib/executor/zap/pkg/runner/runner.go]: removed this as well

**Pull request merged**